src/main/java/com/amazonaws/services/neptune/cluster/AddCloneTask.java

```diff
@@ -378,7 +382,44 @@ private void createInstance(String name,
             requestBuilder = requestBuilder.engineVersion(engineVersion);
         }
 
-        DBInstance targetDbInstance = neptune.createDBInstance(requestBuilder.build()).dbInstance();
+        // Retry configuration
+        int maxRetries = 3;
+        long initialBackoffMillis = 1000; // 1 second
+        DBInstance targetDbInstance = null;
+        CreateDbInstanceRequest request = requestBuilder.build();
+        
+        // Retry loop with exponential backoff
+        for (int attempt = 0; attempt <= maxRetries; attempt++) {
+            try {
+                targetDbInstance = neptune.createDBInstance(request).dbInstance();
+                // If we get here, the request was successful
+                break;
+            } catch (NeptuneException e) {
```

Comment:
This is fine as-is. It is possible to enhance this retry logic by bailing out when you encounter an unrecoverable error.

For example, in HTTP we consider 400-499 status codes unrecoverable. They represent an issue on the request side, not on the server side. I'm not sure what the analogous version of that would be here.

Bailing out early for these types of errors will result in a faster failure for the user in cases where the request will always fail.

Reply:
I looked into this and there is a status code included in the exception message of `NeptuneException` which I could parse out, however in the case which motivated this fix (high replication count leading to createInstance failures), the error message contains a status 400 which is misleading. Further, by looking at the code it appears that `NeptuneException.isRetryable()` will always return false no matter the cause of the error.

For now, I would prefer to always retry despite the potential inefficiency, as it's proving infeasible to distinguish a truly unrecoverable error from one worth retrying.